### PR TITLE
Rework Docker image build to multistage

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -49,23 +49,6 @@ ext {
   shouldUseDelombok = project.hasProperty('useDelombok')
 }
 
-task resolveDependencies {
```
Comment (PawelLipski) [inline build.gradle:52]: Was failing on https://github.com/kelloggm/checkerframework-gradle-plugin/issues/109 for some time

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
-#Thu Jul 02 20:00:58 CEST 2020
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-all.zip
+#Fri Jul 17 22:27:36 CEST 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip
```
Comment (mkondratek) [inline gradle/wrapper/gradle-wrapper.properties:2]: :up:

File: .circleci/config.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ version: 2.1
 jobs:
   build:
     docker:
-      - image: gitmachete/intellij-plugin-ci
+      - image: tilius/git-machete-intellij-plugin-ci # gitmachete/intellij-plugin-ci
```
Comment (PawelLipski) [inline .circleci/config.yml:5]: TODO: push the image as `gitmachete/intellij-plugin-ci` + revert the change

File: Dockerfile [hunk 4]
```diff
@@ -74,4 +58,18 @@ RUN set -x \
...
+# xxd            => xxd
+# unzip          => zipinfo
+RUN set -x \
+  && apt-get update \
+  && apt-get install --no-install-recommends -y binutils netcat procps xxd unzip \
+  `# tools necessary to run non-headless UI tests in the screen-less environment of CI` \
+  && apt-get install --no-install-recommends -y libx11-6 libxrender1 libxtst6 xauth xvfb \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY --from=gradle-cache /root/.gradle/ /root/.gradle/
```
Comment (micpiotrowski) [inline Dockerfile:75]: Nice!

File: .circleci/config.yml (hunk 3 above)
Comment (PawelLipski) [reply .circleci/config.yml]: Done

**Pull request merged**